admin_scripts/reposync.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+trap "exit 1" TERM
+export XCAT_TOOLS_TOP_PID=$BASHPID
+
+BASE=___INSTALL_DIR___
+CONF_BASE=$BASE/conf/reposync.d
+LIB=$BASE/libs
+PRG=$( basename "$0" )
+TIMESTAMP=$(date +%F-%s)
+DEBUG=0
+
+# Import libs
+imports=( logging )
+for f in "${imports[@]}"; do
+    srcfn="${LIB}/${f}.sh"
+    [[ -f "$srcfn" ]] || {
+        echo "Failed to find lib file '$srcfn'"
+        exit 1
+    }
+    source "$srcfn"
+done
+
+usage() {
+  cat <<ENDHERE
+$PRG
+    Sync repos listed under $CONF_BASE/NAME.conf
+Usage:
+    $PRG [OPTIONS] [file]
+    where:
+	file is the path to a specific reposync.d/XXX.conf file to source
+        (otherwise sync all repos under $BASE/conf/reposync.d/XXX.conf)
+
+    Conf file format:
+        A .conf file should define the following variables to source:
+        OS=XXX
+        ARCH=XXX
+        REPO_BASE_DIR=XXX
+        REPO_DIR=XXX
+        REPOS=(
+        REPO1
+        REPO2
+        )
+OPTIONS:
+  -h    print help message and exit
+  -f    file to source to configure which repos to sync
+ENDHERE
+}
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        -h|--help) usage; exit 0;;
+        -f|--file) file="$2"; shift;;
+        -d|--debug) DEBUG=1; shift;;
+        *) echo "Unknown parameter passed: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+do_reposync() {
+    [[ $DEBUG -eq 1 ]] && set -x
+    local _cfg_file="$1"
+    source "$_cfg_file"
```

Comment:
Suggest:
* Add a comment that points out that "$REPOS is defined in the cfg_file"